alibi/explainers/integrated_gradients.py

```diff
@@ -543,20 +547,85 @@ def _format_target(target: Union[None, int, list, np.ndarray],
-            target = [target for _ in range(nb_samples)]
-        elif isinstance(target, list) or isinstance(target, np.ndarray):
-            target = [t.astype(int) for t in target]
+            target = np.array([target for _ in range(nb_samples)])
+        elif isinstance(target, list):
+            target = np.array(target)
+        elif isinstance(target, np.ndarray):
+            pass
         else:
             raise NotImplementedError
 
     return target
 
 
+def _check_target(output_shape: Tuple,
+                  target: Optional[np.ndarray],
+                  nb_samples: int) -> None:
+    """
+    Parameters
+    ----------
+    output_shape
+        Output shape of the tensorflow model
+    target
+        Target formatted as np array target.
+    nb_samples
+        Number of samples in the batch.
+
+    Returns
+    -------
+    None
+
+    """
+
+    if target is not None:
+
+        if not (target.dtype == int):
+            raise ValueError("Targets must be integers")
+
+        if target.shape[0] != nb_samples:
+            raise ValueError(f"First dimension in target must be the same as nb of samples. "
+                             f"Found target 1st dimension: {target.shape[0]}; nb samples: {nb_samples}")
+
+        if len(target.shape) > 2:
+            raise ValueError("Target must be a rank-1 or a rank-2 tensor. If target is a rank-2 tensor, "
+                             "each column contains the index of the corresponding dimension "
+                             "in the model's output tensor.")
+
+        if len(output_shape) == 1:
+            out_rank, target_rank = 1, len(target.shape)
+            tmax, tmin = target.max(axis=0), target.min(axis=0)
+
+            if tmax > 1:
+                raise ValueError(f"Target value {tmax} out of range for output shape = 1 ")
+
+        elif len(output_shape) == 2:
+            out_rank, target_rank = 1, len(target.shape)
+            tmax, tmin = target.max(axis=0), target.min(axis=0)
+
+            if (output_shape[-1] > 1 and (tmax >= output_shape[-1]).any()) or (output_shape[-1] == 1 and tmax > 1):
+                raise ValueError(f"Target value {tmax} out of range for output shape {output_shape} ")
+
+        else:
+            out_rank, target_rank = len(output_shape[1:]), target.shape[-1]
+            tmax, tmin = target.max(axis=0), target.min(axis=0)
+
+            if (tmax >= output_shape[1:]).any():
+                raise ValueError(f"Target value {tmax} out of range for output shape {output_shape} ")
+
+        if (tmin < 0).any():
+            raise ValueError(f"Negative value {tmin} for target. Targets represent positional "
+                             f"arguments and cannot be negative")
+
+        if out_rank != target_rank:
+            raise ValueError(f"The last dimension of target  must match the rank of the model's output tensor. "
+                             f"Found target last dimension: {target_rank}; model's output rank: {out_rank}")
```

Comment:
Should this be done as the very first check as it could mess up everything immediately if not passed right?

Although I'm confused since it seems `target_rank` is not always `len(target.shape)` as it's defined as `target.shape[-1]` for rank>2 outputs...

Reply:
The case with output_shape > 2 is different because the last dimension of the target must match the rank of the output tensor (excluding batch dimension )